src/py/flwr/client/app.py

```diff
@@ -196,7 +227,12 @@ class `flwr.client.Client`.
                 if task_ins is None:
                     time.sleep(3)  # Wait for 3s before asking again
                     continue
-                task_res, sleep_duration, keep_going = handle(client, task_ins)
+
+                # Instantiate the client
+                client_like: ClientLike = client_fn("-1")
```

Comment:
We should do this as late in the process as possible. If, for example, we get a `Reconnect` message here, we don't ever need the `Client` instance. In addition to that, passing the `client_fn` to `message_handler.handle` would enable us to use `handle` in `start_simulation`.

Reply:
I moved it inside `handle()`.